typescript/infra/src/agents/index.ts

```diff
@@ -113,12 +120,20 @@ export abstract class AgentHelmManager {
         chains: this.config.environmentChainNames.map((name) => ({
           name,
           disabled: !this.config.contextChainNames.includes(name),
-          connection: { type: this.config.connectionType },
+          connection: { type: this.connectionType(name) },
         })),
       },
     };
   }
 
+  connectionType(chain: ChainName): AgentConnectionType {
+    if (chainMetadata[chain].protocol == ProtocolType.Sealevel) {
+      return AgentConnectionType.Http;
+    }
+
+    return this.config.connectionType;
```

Comment:
agree this and https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2412#discussion_r1238776233 are good ideas but I'd really like to avoid bigger changes as imo it starts increasing the desired scope / goal of getting something working with Solana (i.e. what i say here https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2412#discussion_r1245210641)

Curious how strongly you feel with these